### DISCUSSION TITLE
How can I insert a breakpoint in Pelican's code and truly stop it?

### DISCUSSION THREAD

**Discussion by offbyone** [getpelican/pelican#3067] [Q&A]
I am trying to inspect part of the pelican core code, and I dropped a `breakpoint()` where I wanted to look, but running `pelican -D` seems to do something to stdin/stdout; I couldn't interact with the `pdb` prompt at all, and the generator just kept going.

<img width="294" alt="image" src="https://user-images.githubusercontent.com/181693/205147509-6d08e2e7-4db9-41b5-a70e-4fe6e199bda2.png">

Is there some way to effectively insert breakpoints in code running in the Pelican generator? If so, how?

**Comment by ghost** [comment]
Set the environment variable `TERM` to `"dumb"`. That disables (among other things) the `⠙ Generating...` status display, so you can actually use pdb.
